manifests/value.pp

```diff
@@ -57,7 +57,8 @@
   validate_re($type, '^\w+',
     'type parameter must not be empty but it is type => "$type"')
 
-
+  # normalize the case so that our defined() check is more consistent
+  $key_normalized = downcase($key)
```

Comment:
The only issue is in setting the value. It should carry the case that the user sets the resource with when the manifest results in creating the resource. Other than that, 👍